api/handlers/refererhandler.py

```diff
@@ -378,7 +383,7 @@ def _prepare_batch(self, fileinfo, analysis):
             filepath = os.path.join(data_path, util.path_from_hash(f['hash']))
             if os.path.exists(filepath): # silently skip missing files
                 targets.append((filepath,
-                                util.sanitize_string_to_filename(analysis['label']) + '/' + f['name'],
+                                util.sanitize_string_to_filename(analysis['label']) + '/' + self._input_or_output(f) + '/'+ f['name'],
```

Comment:
It's not a huge deal, but this can be written inline as 
`util.sanitize_string_to_filename(analysis['label']) + '/' + 'input' if f.get('input') else 'output' + '/'+ f['name'],`